src/Components/Server/src/Builder/EndpointBuilderComponentExtensions.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Components.Server;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extensions for <see cref="IEndpointConventionBuilder"/>.
+    /// </summary>
+    public static class EndpointConventionBuilderComponentsExtensions
+    {
+        /// <summary>
+        /// Adds <typeparamref name="TComponent"/> to the list of components registered with this <see cref="ComponentHub"/> instance.
+        /// </summary>
+        /// <typeparam name="TComponent">The component type.</typeparam>
+        /// <param name="builder">The <see cref="IEndpointConventionBuilder"/>.</param>
+        /// <param name="selector">A CSS selector that identifies the DOM element into which the <typeparamref name="TComponent"/> will be placed.</param>
+        /// <returns>The <paramref name="builder"/>.</returns>
+        public static IEndpointConventionBuilder AddComponent<TComponent>(this IEndpointConventionBuilder builder, string selector)
```

Comment:
Should `builder` be a strong type instead of the interface? This will offer these suggestions on all of the builders.

Reply:
Yeah, but that has its drawbacks too. If we wrap the builder then other conventions can't reliably downcast it to RouteEndpointConventionBuilder and you can't do things based on Order or Pattern.

I think its fine for now to leave it as it is. I have an email thread where I lay out the issues and discuss the pros/cons. I can file an issue to track it for preview4

Reply:
It looks like this might be your solution to manually registering signalR. I think it's a fine choice to leave this as-is for now 👍

Reply:
Yeah.

Reply:
> It looks like this might be your solution to manually registering signalR. I think it's a fine choice to leave this as-is for now 👍

That too, but I'm just too tired to pull thoughts out of my head 😄